### ISSUE TITLE
Perf stats gathering doesn't work in containerized execution

### ISSUE THREAD

**Issue by wenottingham** [ansible/awx#5283]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
Due to the way that it uses cgroups, it doesn't work in containers (nested cgroups with different Linux versions) We need to solve this in a proper manner.

##### ENVIRONMENT
* AWX version: current

**Comment by ryanpetrello** [comment]
Additionally, we'll have to solve how we package the actual performance plugin in the 2.10 world (likely as part of https://github.com/ansible/awx/issues/7058).

**Comment by wenottingham** [comment]
Between http://docs.podman.io/en/latest/markdown/podman-stats.1.html and native OCP features, we may be able to do this without a plugin.

**Comment by AlanCoding** [comment]
Yes, it is expected this will be available in the default execution environment.

https://github.com/ansible/awx/blob/aceb8229ba355068300ace4112a07d9be60b7703/requirements/collections_requirements.yml#L12

https://github.com/ansible-collections/ansible.posix/blob/main/plugins/callback/cgroup_perf_recap.py

> Between http://docs.podman.io/en/latest/markdown/podman-stats.1.html and native OCP features, we may be able to do this without a plugin.

If we get stats from podman, they are not going to be in the same form that we used to get from the callback plugin. The callback plugin runs inside the ansible-playbook process, inside the container.

If you started building something new from this `podman stats` command, I'm worried that the process management would be completely different.

**Comment by wenottingham** [comment]
I expect it would be different, but I also feel that building something on it will be better going forwards than relying on getting the perf plugin to work universally.

**Comment by AlanCoding** [comment]
The primary loop for `ansible-runner` is `while child.isalive():`.

https://github.com/ansible/ansible-runner/blob/c2a960193484ce57bfb462749b671088c2a3e97c/ansible_runner/runner.py#L213

From what I'm seeing of `podman stats`, this works to get the present resource consumption of a running container. The `child` in this case would be the `podman run` subprocess.

In terms of technical design, would we be adding logic inside this loop to run the stats command? That would require applying a name to the container when we start it, but this need is shared by the issue https://github.com/ansible/ansible-runner/issues/506, so needs work, but no issue there.

The interval could be somewhat controlled by not running that command every loop, but there is a minimum interval, which is whatever rate pexpect returns control to us. Time taken to run the stats command, could make other things more laggy, like certain callback methods.

A lot of the test failures we have to weed through (for execution environments) are performance stats failures. Even if we implement this new solution, I expect the majority of those tests will have to be rewritten, because they tend to be highly implementation-specific, going as far as looking for specific things in specific files on the server. Ping @matburt, I want him to be aware of the specs and scope of what is being discussed.

**Comment by AlanCoding** [comment]
Filed an issue in ansible-runner.

I have a PR up for related work that adds container names, which is needed given the design I outlined.

**Comment by AlanCoding** [comment]
I know that @chrismeyersfsu started looking at this and was coming up with some ideas. Those ideas have not been documented to my knowledge. This is a pretty major chunk of work.

**Comment by kdelee** [comment]
afaik this has been punted for the time being and we may land after main containerization stuff
